public/app/features/alerting/unified/utils/rule-form.ts

```diff
@@ -307,6 +335,23 @@ export function getContactPointsFromDTO(ga: GrafanaRuleDefinition): AlertManager
   return routingSettings;
 }
 
+function getEditorSettingsFromDTO(ga: GrafanaRuleDefinition) {
+  // we need to check if the feature toggle is enabled as it might be disabled after the rule was created with the feature enabled
+  if (config.featureToggles.alertingQueryAndExpressionsStepMode) {
+    if (ga.metadata?.editor_settings) {
+      return {
+        simplifiedQueryEditor: ga.metadata.editor_settings.simplified_query_and_expressions_section,
+      };
+    } else {
+      return {
+        simplifiedQueryEditor: false,
+      };
+    }
+  } else {
+    return undefined;
+  }
+}
```

Comment:
Nit, refactor to early return and remove `else`

```ts
function getEditorSettingsFromDTO(ga: GrafanaRuleDefinition) {
  // we need to check if the feature toggle is enabled as it might be disabled after the rule was created with the feature enabled
  if (!config.featureToggles.alertingQueryAndExpressionsStepMode) {
    return undefined;
  }

  if (ga.metadata?.editor_settings) {
    return {
      simplifiedQueryEditor: ga.metadata.editor_settings.simplified_query_and_expressions_section,
    };
  }

  return {
    simplifiedQueryEditor: false,
  };
}
```